{% extends base.html %}
{%block head%}

{%end%}

{% block content%}

<div>
<div class="row">
  	<h3>Welcome to the Codon Compression web tool</h3>
</div>

<div class="row" style="padding:10px 10px 30px 0px"><p style="font-size:16px;">Saturation mutagenesis is employed in protein engineering and genome-editing efforts to generate libraries that span amino acid design space. Our codon compression web tool allows users to compress degenerate codons in a user specified manner. The output provided will be a unique collection of codons for any desired amino acids. <br><br>

This tool’s purpose is to eliminate undesired amino acids, stop codons and redundancy (or bias) from the generated diversity, as occurs in the commonly used NNK codon (that includes 32 codons for 20 amino acids and includes also stop codon). <br><br>

The resulting compressed codons are optimized for the target organism codon usage, and custom genetic and usage tables can be uploaded for exotic organisms or when non-natural amino acids are in use. <br><br>

This tool contains 2 different compression approaches:<br><br> 

<u><a href="/dynamcc_0/"><strong>DYNAMCC_0</strong></a></u>:
This algorithm calculates the minimal set of degenerate codons for any desired set of amino acids, without redundancy. Note that the compressed codons may be ranked according to codon usage, or according to their usage rank, which may result in different results.<br><br>
 
For example: to include all amino acids, without redundancy and without any stop codon, optimized for <em>E. coli</em>, the compressed codon list is: VHG, NRT, WTT and TGG<br><br>

<u><a href="/dynamcc_R/"><strong>DYNAMCC_R</strong></a></u>:
This algorithm helps in finding compressed codons with various levels of redundancy. Due to the large solution space, the algorithm compresses the “best List” according to usage that constitutes a single codon for each amino acid, and then adds more codons ordered by their specific usage, until complete redundancy is achieved.<br><br>
For example: to include all amino acids, without any stop codon, optimized for <em>E. coli</em>, the compressed and fully redundant codon list is: NYA, NBG, VRA, NNY and VAG<br><br>

Additionally, we have provided the capability to "explode" a set of compressed codons:<br><br>

<u><a href="/dynamcc_d/"><strong>DYNAMCC_D</strong></a></u>:
Here the user can define the Hamming distance between the wild type codon and the mutated ones. The aim of this algorithm is to further reduce library size, while still presenting relevant results. The rationale is that it has been shown by us and others that in many protein engineering studies the best performing mutant, (or the highest peak in the mutational landscape) requires more than a single base change in the codon1,2. Alternatively, researchers who are interested in SNP research, such as in the case of cancer and other genetic diseases, may want to design their libraries to include a single base change only.
<br><br>

<u><a href="/exploder/"><strong>Codon Exploder</strong></a></u>: This tools allows you to visualize the mapping of compressed codons to the corresponding collection of true codons. The usage and ranking of the "exploded codons" will be determined based on the selected organism and displayed in the output table.<br><br>
 
If you find this tool helpful in your research, <strong>please cite our paper:</strong><br>
<a href="http://pubs.acs.org/doi/abs/10.1021/sb500282v">Pines G., Pines, A., Garst, A.G., Zeitoun, R.I., Lynch, S.A., and Gill, R.T. Codon Compression Algorithms for Saturation Mutagenesis. ACS Synth. Biol. 4, 604–614 (2015).</a><br><br>

<div class="row" style="margin:0 0 0 0;">
	<div class="col-md-3" style="margin:10px 0 0 0;padding:0 0 0 0;font-size:16px;">Find out more about ongoing projects in the</div>
	<div class="col-md-4" style="margin:0 0 0 10px;padding:0 0 0 0;"><a href="http://www.gillgroup.org/" class="navbar-brand lab_logo"></a></div>
</div>
</p>

</div>


</div>

{%end%}
